homeassistant/components/sensor/wirelesstag.py

```diff
@@ -0,0 +1,168 @@
+SENSOR_HUMIDITY = 'humidity'
+SENSOR_MOISTURE = 'moisture'
+SENSOR_LIGHT = 'light'
+
+SENSOR_TEMPERATURE_F_ICON = 'temperature-fahrenheit'
+
+SENSOR_TYPES = {
+    SENSOR_TEMPERATURE: {
+        'unit': TEMP_CELSIUS,
+        'icon': 'temperature-celsius',
+        'attr': 'temperature'
+    },
+    SENSOR_HUMIDITY: {
+        'unit': '%',
+        'icon': 'water-percent',
+        'attr': 'humidity'
+    },
+    SENSOR_MOISTURE: {
+        'unit': '%',
+        'icon': 'water',
+        'attr': 'moisture'
+    },
+    SENSOR_LIGHT: {
+        'unit': 'lux',
+        'icon': 'brightness-6',
+        'attr': 'light'
+    }
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_ENTITY_NAMESPACE, default=DEFAULT_ENTITY_NAMESPACE):
+        cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    platform = hass.data.get(WIRELESSTAG_DOMAIN)
+    sensors = []
+    tags = platform.tags
+    for _, tag in tags.items():
+        for sensor_type in config.get(CONF_MONITORED_CONDITIONS):
+            if sensor_type in WirelessTagSensor.allowed_sensors(tag):
+                sensors.append(WirelessTagSensor(
+                    platform, tag, sensor_type, hass.config))
+
+    add_devices(sensors, True)
+
+
+class WirelessTagSensor(WirelessTagBaseSensor):
+    """Representation of a Sensor."""
+
+    @classmethod
+    def allowed_sensors(cls, tag):
+        """Return array of allowed sensor types for tag."""
+        all_sensors = SENSOR_TYPES.keys()
+        sensors_per_tag_type = {
+            WIRELESSTAG_TYPE_13BIT: [
+                SENSOR_TEMPERATURE,
+                SENSOR_HUMIDITY],
+            WIRELESSTAG_TYPE_WATER: [
+                SENSOR_TEMPERATURE,
+                SENSOR_MOISTURE],
+            WIRELESSTAG_TYPE_ALSPRO: [
+                SENSOR_TEMPERATURE,
+                SENSOR_HUMIDITY,
+                SENSOR_LIGHT]
+        }
+
+        tag_type = tag.tag_type
+        return (
+            sensors_per_tag_type[tag_type] if tag_type in sensors_per_tag_type
+            else all_sensors)
+
+    def __init__(self, api, tag, sensor_type, config):
+        """Constructor with platform(api), tag and hass sensor type."""
+        super().__init__(api, tag)
+
+        self._sensor_type = sensor_type
+        self._tag_attr = SENSOR_TYPES[self._sensor_type]['attr']
+        self._unit_of_measurement = SENSOR_TYPES[self._sensor_type]['unit']
+        self._icon = 'mdi:{}'.format(SENSOR_TYPES[self._sensor_type]['icon'])
+        if sensor_type == SENSOR_TEMPERATURE and not config.units.is_metric:
+            self._icon = 'mdi:{}'.format(SENSOR_TEMPERATURE_F_ICON)
+        self.define_name(self._tag.name)
+
+        # sensor.wirelesstag_bedroom_temperature
+        self._entity_id = '{}.{}_{}_{}'.format('sensor', WIRELESSTAG_DOMAIN,
```

Comment:
Why would you define an entity id? Way better to define a unique ID if you can and allow the user to manage entity id via the entity registry.